pkg/resources/manager/manager.go

```diff
@@ -165,6 +183,12 @@ func (r *resourceManager) initialize(ctx context.Context, machines []*api.Machin
 
 	r.ctx = ctx
 
+	totalExistingVMCount := len(machines)
+	if r.vmCountLimit != 0 && totalExistingVMCount >= r.vmCountLimit {
+		r.log.Info("VM limit is already reached", "Limit", r.vmCountLimit, "Existing count", totalExistingVMCount)
+	}
+	r.vmCount = totalExistingVMCount
+
```

Comment:
i recommend integrate vm limit count into calculation of machine class quantity. It can help us improve scheduling.

Reply:
Done